### PR TITLE
Fix phase started bug

### PR DESCRIPTION
**Pull request by edumoreira1506**
## References

> Related issue #4011 

## Objectives

> Fix ArgumentError in Legislation::ProcessesController#allegations

## Visual Changes

Peek 2020-06-24 18-18

### CODE REVIEW

File: app/models/legislation/process/phase.rb [hunk 1]
```diff
@@ -10,7 +10,7 @@ def enabled?
   end
 
   def started?
-    enabled? && Date.current >= @start_date
+    return enabled? && Date.current >= @start_date unless @start_date.nil?
```
Comment (lairtonlelis) [inline app/models/legislation/process/phase.rb:13]: `return` is unnecessary in this case, ruby has an implicit return. And regarding the use of `unless` for the nil check, a simple `if` would make the code more readable, like this:
```suggestion
    enabled? && Date.current >= @start_date if @start_date
```

File: spec/models/legislation/process/phase_spec.rb [hunk 2]
```diff
@@ -49,6 +49,11 @@
       process.update!(debate_start_date: Date.current - 2.days,
                       debate_end_date: Date.current - 1.day)
       expect(process.debate_phase.started?).to be true
+
+      # nil
+      process.update(!debate_start_date: nil,
```
Comment (lairtonlelis) [inline spec/models/legislation/process/phase_spec.rb:54]: There is a syntax error here. Perhaps you meant to use `update!`?
```suggestion
      process.update!(debate_start_date: nil,
```

Comment (edumoreira1506) [reply spec/models/legislation/process/phase_spec.rb]: Yep, my bad, thank you.

File: app/models/legislation/process/phase.rb (hunk 1 above)
Comment (edumoreira1506) [reply app/models/legislation/process/phase.rb]: Fixed.

**Comment by edumoreira1506** [comment]
Could you review @decabeza ?

**Comment by javierm** [comment]
@edumoreira1506 Thanks for your contribution! :pray:

Before reviewing, could you comment what your thoughts about adding validation rules (as @decabeza suggested in issue #4011) are? Personally I like the idea a lot, but I haven't investigated the issue so I'm not sure whether we could implement it by changing these lines in the `Legislation::Process` or this approach would make the applcation work unexpectedly in some places :thinking::

```ruby
  validates :debate_start_date, presence: true, if: :debate_end_date?
  validates :debate_end_date, presence: true, if: :debate_start_date?
  validates :draft_start_date, presence: true, if: :draft_end_date?
  validates :draft_end_date, presence: true, if: :draft_start_date?
  validates :allegations_start_date, presence: true, if: :allegations_end_date?
  validates :allegations_end_date, presence: true, if: :allegations_start_date?
  validates :proposals_phase_end_date, presence: true, if: :proposals_phase_start_date?
```

**Comment by javierm** [comment]
@edumoreira1506 Any comments about the validation rules I mentioned?

**Comment by javierm** [comment]
Closing due to lack of response.

**Pull request closed without merging**